Switch instrument search to use the POST api url.

tjader opened the pull request:
This should resolve #113.

Qluxzz commented:
The return models of the old and new URL are not compatible so this is also a breaking change.

Running the test suite `python -m unittest` confirms this.
We'd need to update the different return models to match the new ones.

tjader commented:
The search results all have the same output, so I merged them to use the same model. Also fixed a small issue in the transactions model (and added some bits into .gitignore that shouldn't be committed)

Qluxzz commented:
LGTM, thanks for updating the tests and models!

jonatanj commented on avanza/models/search_result.py:
```diff
@@ -0,0 +1,67 @@
+class SearchPrice(BaseModel):
+    # Most values are returned as strings.
+    # null values are returned as "null"
+    last: str|None
```
Note that this code makes the library incompatible with Python 3.9 (the minimum version stated in setup.py). See https://docs.python.org/3.10/whatsnew/3.10.html#pep-604-new-type-union-operator .
